src/objects/basic_npc.py

```diff
@@ -111,17 +92,11 @@ def take_damage(self, damage):
         if self.hp <= 0:
             self.alive = False
 
-    def shoot_bullet(self, collision_sensor, bullet_id):
+    def shoot_bullet(self, offset, collision_sensor):
         self.bullet_ready = False
         self.prev_shot_time = time.time()
-        # offset: radius of the shooter to avoid bullet collision with the shooter itself
-        offset = self.radius + self.bullet_radius
-        bullet_pos = self.pos + offset * self.dir
-        self.game_stats.track_bullet_fired()
-        return Bullet(bullet_pos, self.dir, self.damage, self.bullet_radius, collision_sensor, bullet_id)
```

Comment:
the removed attributes are correct, offset is calculated by the radii of the objects